homeassistant/components/velux.py

```diff
@@ -0,0 +1,71 @@
+"""
+Connects to VELUX KLF 200 interface.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/scene.velux/
+
+"""
+
+import logging
+import asyncio
+
+import voluptuous as vol
+
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (CONF_HOST, CONF_PASSWORD)
+
+DOMAIN = "velux"
+SUPPORTED_DOMAINS = ['scene']
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['pyvlx==0.1.3']
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Set up the velux component."""
+    from pyvlx import PyVLXException
+    try:
+        if DOMAIN not in hass.data:
+            hass.data[DOMAIN] = VeluxModule(hass, config)
+            yield from hass.data[DOMAIN].async_start()
+
+    except PyVLXException as ex:
+        _LOGGER.exception("Can't connect to velux interface: %s", ex)
+        hass.data[DOMAIN] = None
```

Comment:
Hmm, that's a legit point about the clean-up. At first glance it looks like that it will be used for the check in the platform setup.

But we should delete the key in dict instead of setting it to `None`.